src/project/manifest.rs

```diff
@@ -182,17 +182,17 @@ impl<'de> Deserialize<'de> for TargetSelector {
 pub struct TargetMetadata {
     /// Target specific dependencies
     #[serde(default)]
-    #[serde_as(as = "IndexMap<_, DisplayFromStr>")]
+    #[serde_as(as = "IndexMap<_, PickFirst<(_, DisplayFromStr)>>")]
     pub dependencies: IndexMap<String, NamelessMatchSpec>,
 
     /// The host-dependencies of the project.
     #[serde(default, rename = "host-dependencies")]
-    #[serde_as(as = "Option<IndexMap<_, DisplayFromStr>>")]
+    #[serde_as(as = "Option<IndexMap<_, PickFirst<(_, DisplayFromStr)>>>")]
     pub host_dependencies: Option<IndexMap<String, NamelessMatchSpec>>,
 
     /// The build-dependencies of the project.
     #[serde(default, rename = "build-dependencies")]
-    #[serde_as(as = "Option<IndexMap<_, DisplayFromStr>>")]
+    #[serde_as(as = "Option<IndexMap<_, PickFirst<(_, DisplayFromStr)>>>")]
```

Comment:
What does `PickFirst` do?

Reply:
It picks the first serialization that succeeds. This can now be a map or a string. 
https://docs.rs/serde_with/latest/serde_with/struct.PickFirst.html